### ISSUE TITLE
Allow the use of private IP even though ECS task has a public IP

### ISSUE THREAD

**Issue by noppaz** [dask/dask-cloudprovider#287]
When initiating an ECS- or FargateCluster we have the option to `fargate_use_private_ip`. Depending on this boolean, the cluster object's scheduler_address will be either private or public.

This is limiting depending on the use cases:
1. Either, if we use `fargate_use_private_ip=True` we'll have no means of access internet resources without a NAT gateway. This includes the initial pull of the task's ECR image.
2. Or, we use `fargate_use_private_ip=False` and the communication between the client creating the cluster and the scheduler will be done on the public IP even though the client could be inside the same VPC. For security reasons keeping that communication within the VPC would be preferable.

My specific use case is using Prefect's [DaskExecutor](https://docs.prefect.io/orchestration/flow_config/executors.html#using-a-temporary-cluster) which is a separate ECS task with a client creating a Dask cluster from within the same VPC. Other potential use cases could be a Jupyter instance on ECS or EC2 within the same VPC as the cluster, with this we could limit the internet facing ports of the VPC to only Jupyter ones and potentially the cluster scheduler's dashboard (8687) or limit inbound internet access completely.

From my initial understanding of the code I think this could be done either inside dask_cloudprovider by pretending that we didn't want a public IP only when setting that address [here](https://github.com/dask/dask-cloudprovider/blob/main/dask_cloudprovider/aws/ecs.py#L203). This would presumably require an additional boolean argument e.g., `connect_to_private_ip`, or you'd to make more substantial changes to dask/distributed.

Basically, what I'd like to be able to do is something like
```
# Current
>>> cluster = FargateCluster(n_workers=1, fargate_use_private_ip=False)
>>> print(cluster)
FargateCluster(<name>, 'tcp://<public-ip>:8786', workers=1, threads=0, memory=0 B)

# Suggestion
>>> cluster = FargateCluster(n_workers=1, fargate_use_private_ip=False, connect_to_private_ip=True)
>>> print(cluster)
FargateCluster(<name>, 'tcp://<private-ip>:8786', workers=1, threads=0, memory=0 B)
```

I'd like to bring this up for discussion as I think its a useful enhancement. I've also seen other's discussing the matter in the Prefect Slack community. I'd be happy to give my best to contribute to a potential change once we have some orientation.

**Comment by noppaz** [comment]
Hey Jacob, thanks for looking over this idea.

I agree it sounds a bit like I want to have the cake and eat it too. Trying to keep things simple while not having to open up the VPC ingress was the goal here. From my point of view it could people set up and utilise dask distributed in a securer way more easily. But yes, I can agree that it might be better to force people to go the NAT gateway route for this circumstance.

**Comment by jacobtomlinson** [comment]
Using a NAT to allow internet access from within a private VPC is the AWS recommended best practice, so I think it's best for us to stick with that.
